### ISSUE TITLE
Incorrectly detecting Utility class available

### ISSUE THREAD

**Issue by mnunezdm** [forcedotcom/salesforcedx-vscode-slds#32]
### Summary

When using pseudoclasses, the extension still recommends using a utility class, which results in incorrect selection of the desired tag.

### Steps To Reproduce:

1.  Create a new CSS entry with a pseudoclass, for example, :last-child
1.  An error appears on screen recommending to use the utility class

### Expected result

Nothing

### Actual result

![image](https://user-images.githubusercontent.com/10410852/88531427-1ee49e80-d003-11ea-88cf-c2f76a1052c7.png)

### Additional information

_Feel free to attach a screenshot_.

**VS Code Version**: 1.47.2

**SFDX CLI Version**:

@oclif/plugin-autocomplete 0.1.5 (core)
@oclif/plugin-commands 1.3.0 (core)
@oclif/plugin-help 3.0.1 (core)
@oclif/plugin-not-found 1.2.4 (core)
@oclif/plugin-plugins 1.7.10 (core)
@oclif/plugin-update 1.3.10 (core)
@oclif/plugin-warn-if-update-available 1.7.0 (core)
@oclif/plugin-which 1.0.3 (core)
@salesforce/sfdx-trust 3.0.7 (core)
analytics 1.12.0 (core)
generator 1.1.3 (core)
salesforcedx 49.2.3 (core)
├─ salesforce-alm 49.3.0 (core)
├─ custom-metadata 1.0.7 (core)
├─ salesforcedx-templates 48.32.0 (core)
└─ @salesforce/sfdx-plugin-lwc-test 0.1.7 (core)
sfdx-cli 7.66.2 (core)
sfdx-wry-plugin 0.0.11

**OS and version**: WSL 2 Running in Windows 10

**Comment by ambreeen** [comment]
Thanks for reporting. Fixed in the latest release.

**Comment by billyma** [comment]
@mnunezdm - I tried these pseudo classes: `first-child`, `last-child`, `nth-child`, `nth-type`. I was not able to reproduce the issue you are mentioning. I may not have cover all the scenarios. It would be great if you can provide the accompanying markup that can reproduce the issue. Thanks.

**Comment by mnunezdm** [comment]
Since last update im not able to get the slds validator running, so I am unable to replicate the issue, will troubleshoot and see what happened there and then reach you with evidences.

Did you try in a LWC or in a Aura? in my case was a LWC maybe has something to do with that

**Comment by mnunezdm** [comment]
Hi, I validated and replicated this error in both LWC & Aura components,

In order to view the error you must add the pseudoclass to an existing & used class.

For example if you create a new class in the css and add the pseudoclass with the margin-bottom, you dont receive any error.

![image](https://user-images.githubusercontent.com/10410852/90501127-256ebd80-e14c-11ea-9e69-a337dae4fbd0.png)

Also, in order to replicate, you can use this component

```js
import { LightningElement, api, track } from 'lwc';

export default class TestPseudo extends LightningElement {
  items = [
    {
      id: 1,
      value: 'foo',
    },
    {
      id: 2,
      value: 'bar',
    },
  ]
}
```

```html
<template>
  <ul class="items">
    <li class="item" for:each={items} for:item="item" key={item.id}>
      {item}
    </li>
  </ul>
</template>
```

```css
.items {
  background: red;
}

.item:last-child {
  margin-bottom: 0;
}
```

**Comment by billyma** [comment]
Let me see if I understood it correctly.  In the example provided, validator should not flag this declaration, since the element is within a for loop. Applying the recommendation to the element would result in a different behavior.

**Comment by mnunezdm** [comment]
yeah, thats it, I normally use pseudo selectors when i need to change a subset of elements inside an group of elements

![examplePseudo](https://user-images.githubusercontent.com/10410852/90624009-97600900-e217-11ea-9621-e3d8493d7c2a.gif)

For example, in that case, I want all the elements to have margin bottom of 16px but the last element I don't want it to have the margin bottom, so I use the last-child to override the margin and set it to 0.

We also have to validate if this affects to pseudo elements like ::before and ::after

**Comment by billyma** [comment]
I updated the pull request with better handling of psuedo-classes.
